src/main/java/org/threadly/db/aurora/AuroraDataSource.java

```diff
@@ -0,0 +1,321 @@
+   */
+  public int getPort() {
+    return this.port;
+  }
+  
+  /**
+   * Sets the user ID.
+   * 
+   * @param userID
+   *          the User ID
+   */
+  public void setUser(String userID) {
+    this.user = userID;
+  }
+  
+  /**
+   * Returns the configured user for this connection
+   * 
+   * @return the user for this connection
+   */
+  public String getUser() {
+    return this.user;
+  }
+  
+  /**
+   * Sets the database name.
+   * 
+   * @param dbName
+   *          the name of the database
+   */
+  public void setDatabaseName(String dbName) {
+    this.databaseName = dbName;
+  }
+  
+  /**
+   * Gets the name of the database
+   * 
+   * @return the name of the database for this data source
+   */
+  public String getDatabaseName() {
+    return (this.databaseName != null) ? this.databaseName : "";
+  }
+  
+  /**
+   * Sets the server name.
+   * 
+   * @param serverName
+   *          the server name
+   */
+  public void setServerName(String serverName) {
+    this.hostName = serverName;
+  }
+  
+  /**
+   * Returns the name of the database server
+   * 
+   * @return the name of the database server
+   */
+  public String getServerName() {
+    return (this.hostName != null) ? this.hostName : "";
+  }
+  
+  /**
+   * Sets the URL for this connection
+   * 
+   * @param url
+   *          the URL for this connection
+   */
+  public void setURL(String url) {
+    setUrl(url);
+  }
+  
+  /**
+   * Returns the URL for this connection
+   * 
+   * @return the URL for this connection
+   */
+  public String getURL() {
+    return getUrl();
+  }
+  
+  /**
+   * This method is used by the app server to set the url string specified within the datasource deployment descriptor.
+   * It is discovered using introspection and matches if property name in descriptor is "url".
+   * 
+   * @param url
+   *          url to be used within driver.connect
+   */
+  public void setUrl(String url) {
+    this.url = url;
+    this.explicitUrl = true;
+  }
+  
+  /**
+   * Returns the JDBC URL that will be used to create the database connection.
+   * 
+   * @return the URL for this connection
+   */
+  public String getUrl() {
+    if (!this.explicitUrl) {
+      StringBuilder sbUrl = new StringBuilder(AURORA_PREFIX);
+      sbUrl.append("//").append(getServerName()).append(":").append(getPort()).append("/").append(
+                                                                                                  getDatabaseName());
+      return sbUrl.toString();
+    }
+    return this.url;
+  }
+  
+  /**
+   * Creates a connection using the specified properties.
+   * 
+   * @param props
+   *          the properties to connect with
+   * 
+   * @return a connection to the database
+   * 
+   * @throws SQLException
+   *           if an error occurs
+   */
+  protected java.sql.Connection getConnection(Properties props) throws SQLException {
+    final String jdbcUrlToUse = (!this.explicitUrl) ? getUrl() : this.url;
```

Comment:
This if condition is not really optimizing anything, I would just invoke `getUrl()` directly every time.  Probably just nest as the argument inside the `connect` function below.